src/Cli/dotnet/commands/dotnet-workload/clean/WorkloadCleanCommand.cs

```diff
@@ -0,0 +1,132 @@
+            }
+
+            DisplayUninstallableVSWorkloads();
+        }
+
+        /// <summary>
+        /// Print VS Workloads with the same machine arch which can't be uninstalled through the SDK CLI to increase user awareness that they must uninstall via VS.
+        /// </summary>
+        private void DisplayUninstallableVSWorkloads()
+        {
+#if !DOT_NET_BUILD_FROM_SOURCE
+            if (OperatingSystem.IsWindows())
+            {
+                // All VS Workloads should have a corresponding MSI based SDK. This means we can pull all of the VS SDK feature bands using MSI/VS related registry keys.
+                var installedSDKVersionsWithPotentialVSRecords = MsiInstallerBase.GetInstalledSdkVersions();
+                HashSet<string> vsWorkloadUninstallWarnings = new();
+
+                foreach (string sdkVersion in installedSDKVersionsWithPotentialVSRecords)
+                {
+                    try
+                    {
+#pragma warning disable CS8604 // We error in the constructor if the dotnet path is null.
+                        // The below environment expansion should be architecture agnostic for the C:\Program Files path, but only works on windows 7+. https://learn.microsoft.com/en-us/windows/win32/winprog64/wow64-implementation-details?redirectedfrom=MSDN
+                        // In addition, we don't need to worry about when dotnet will be dotnet/x64 (x64 on processor arch of arm64) because MSI install thru VS will only support installs on processor arch, not chip/emulation arch.
+                        string defaultDotnetWinPath = Path.Combine(Environment.ExpandEnvironmentVariables("%ProgramW6432%"), "dotnet");
+
+                        // We don't know if the dotnet installation for the other bands is in a different directory than the current dotnet; check the default directory if it isn't.
+                        var bandedDotnetPath = Path.Exists(Path.Combine(_dotnetPath, "sdk", sdkVersion)) ? _dotnetPath : defaultDotnetWinPath;
+
+                        var workloadManifestProvider = new SdkDirectoryWorkloadManifestProvider(bandedDotnetPath, sdkVersion, _userProfileDir);
+                        var bandedResolver = WorkloadResolver.Create(workloadManifestProvider, bandedDotnetPath, sdkVersion.ToString(), _userProfileDir);
+#pragma warning restore CS8604
+
+                        InstalledWorkloadsCollection vsWorkloads = new();
+                        VisualStudioWorkloads.GetInstalledWorkloads(bandedResolver, vsWorkloads, _cleanAll ? null : new SdkFeatureBand(sdkVersion));
+                        foreach (var vsWorkload in vsWorkloads.AsEnumerable())
+                        {
+                            vsWorkloadUninstallWarnings.Add(string.Format(LocalizableStrings.VSWorkloadNotRemoved, $"{vsWorkload.Key} ({vsWorkload.Value})"));
+                        }
+                    }
+                    catch (Exception ex)
```

Comment:
Thank you for this suggestion. I have made it catch one specific error type which is WorkloadManifestException. Before if you give a bad path or a path to a different SDK that did have manifests then the manifest reader would fail. I was worried about other things that could happen if the path is wrong and tried to account for that better instead of just catching all exceptions.